### ISSUE TITLE
about acc.I cannot reproduce the accuracy given in the paper

### ISSUE THREAD

**Issue by wys2929** [czhaneva/SkeleMixCLR#1]
**Hello, when I run your skelemixCLR code, I cannot reproduce the accuracy given in the paper.(NUT RGB-D60 xview)**
[09.26.22|16:22:14] ********** KNN Result **********
[09.26.22|16:22:14]     50-10: cur-51.91%, best-51.91%
[09.26.22|16:22:14]     50-20: cur-52.46%, best-52.46%
[09.26.22|16:22:14]     50-40: cur-51.64%, best-51.64%
[09.26.22|16:22:14]     100-10: cur-55.85%, best-55.85%
[09.26.22|16:22:14]     100-20: cur-56.26%, best-56.26%
[09.26.22|16:22:14]     100-40: cur-55.36%, best-55.36%
[09.26.22|16:22:14]     150-10: cur-56.41%, best-56.45%
[09.26.22|16:22:14]     150-20: cur-56.30%, best-56.30%
[09.26.22|16:22:14]     150-40: cur-55.49%, best-55.72%

 **My .yaml configuration is as follows**

work_dir: work_dir/pretrain_skeletonclr_multi_swapv2/202209260942/xview
// work_dir: work_dir/singlegpu/1gpu_bz128_pergpu/seed2/Base/spatial_shear_tem_padding/queue_32768/Tem_0.2/with_MLP/skeclr_multi_swap/semantic_swap/64_3_1/st_gcn/agcn_graph/ntu/joint/0_xview_frame64_channel16_feature128_hidden256_epoch300
//<work_dir> / <seed> / <data> / <aug> / <queue> / <Tem> / <MLP> / <arch> / <Param> / <Backbone> / <Graph> / <dataset> / <detail_training>

// feeder
train_feeder: feeder.ntu_feeder.Feeder_multi
train_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/train_position.npy
  frames: [64, ]
  label_path: /home/wuyushan/project1/CrosSCLR/data/NTU-RGB-D/xview/train_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  mmap: True
  
// memory feeder for knn
mem_train_feeder: feeder.ntu_feeder.Feeder_single
mem_train_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/train_position.npy
  label_path: /home/wuyushan/project1/CrosSCLR/data/NTU-RGB-D/xview/train_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  window_size: 64
  center: True
  mmap: True
mem_test_feeder: feeder.ntu_feeder.Feeder_single
mem_test_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/val_position.npy
  label_path: /home/wuyushan/project1/CrosSCLR/data/NTU-RGB-D/xview/val_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  window_size: 64
  center: True
  mmap: True

// model
model: net.skeletonclr_multi_swapv2.SkeletonCLR
model_args:
  base_encoder: net.st_gcn_swap.Model
  pretrain: True
  feature_dim: 128
  queue_size: 32768
  momentum: 0.999
  Temperature: 0.2
  mlp: True
  in_channels: 3
  hidden_channels: 16 
  hidden_dim: 256
  num_class: 60
  dropout: 0.5
  // spa_l, spa_u, tem_l, tem_u, repeat, loss_weight
  swap_cfg:
    swap_params: [[2, 3, 7, 11, 1, 1.],
                  ]
    swap_mode: 'swap'
    spatial_mode: 'semantic'
  graph_args:
    layout: 'ntu-rgb+d'
    strategy: 'spatial'
  edge_importance_weighting: True

// optim
nesterov: False 
weight_decay: 1e-4
base_lr: 0.1
optimizer: SGD
step: [75]

// training
device: [0]
batch_size: 128
test_batch_size: 128
num_epoch: 150  # 300
mining_epoch: 150  # 300
start_epoch: 0

// cross training
view: 'joint'

// log
save_interval: 5
eval_interval: -1

knn_interval: 5
knn_k: [10, 20, 40]
knn_classes: 60
knn_t: 0.1
KNN_show: [50, 100, 150, 200, 250, 300]

I don't know why this is, I only modified epoch=150, miningepoch=150, step=75, repeat=1, and data path, the gpu used is a NVIDIA TITAN, the highest accuracy I get is only about 56.45%, And in your paper "Contrastive Learning from Spatio-Temporal Mixed Skeleton Sequences for Self-Supervised Skeleton-Based Action Recognition", the accuracy of 100 epochs on NTU-RGB-D60 xview dataset should be 81.3%, can you tell me why and tell me how to fix it?

**Comment by wys2929** [comment]
The run statement is
python main.py pretrain_skeletonclr_multi_swapv2 --config config/CrosSCLR/skelemixclr_multi_swap_ntu.yaml

**Comment by czhaneva** [comment]
1. It is recommended to test the accuracy of the pretrained model provided.
2. No drop in learning rate required.
3. The accuracy you reported is the KNN result, it is recommended to test the accuracy with other evaluation protocols (linear evaluation, finetune evaluation)
4. No mining strategy required.

**Comment by wys2929** [comment]
**Thank you for your prompt reply, I redrew the yaml file, as follows**

work_dir: work_dir/pretrain_skeletonclr_multi_swapv2/202210011046/xview

// feeder
train_feeder: feeder.ntu_feeder.Feeder_multi
train_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/train_position.npy
  frames: [64, ]
  label_path: /home/xxx/project1/data/NTU-RGB-D/xview/train_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  mmap: True
  
// memory feeder for knn
mem_train_feeder: feeder.ntu_feeder.Feeder_single
mem_train_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/train_position.npy
  label_path: /home/xxx/project1/data/NTU-RGB-D/xview/train_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  window_size: 64
  center: True
  mmap: True
mem_test_feeder: feeder.ntu_feeder.Feeder_single
mem_test_feeder_args:
  data_path: data/Base/NTU60_frame64/xview/val_position.npy
  label_path: /home/xxx/project1/data/NTU-RGB-D/xview/val_label.pkl 
  shear_amplitude: 0.5
  temperal_padding_ratio: 6
  window_size: 64
  center: True
  mmap: True

// model
model: net.skeletonclr_multi_swapv2.SkeletonCLR
model_args:
  base_encoder: net.st_gcn_swap.Model
  pretrain: True
  feature_dim: 128
  queue_size: 32768
  momentum: 0.999
  Temperature: 0.2
  mlp: True
  in_channels: 3
  hidden_channels: 16 
  hidden_dim: 256
  num_class: 60
  dropout: 0.5
  // spa_l, spa_u, tem_l, tem_u, repeat, loss_weight
  swap_cfg:
    swap_params: [[2, 3, 7, 11, 1, 1.],
                  ]
    swap_mode: 'swap'
    spatial_mode: 'semantic'
  graph_args:
    layout: 'ntu-rgb+d'
    strategy: 'spatial'
  edge_importance_weighting: True

// optim
nesterov: False 
weight_decay: 1e-4
base_lr: 0.1
optimizer: SGD
step: [300]

// training
device: [0]
batch_size: 128
test_batch_size: 128
num_epoch: 300 
mining_epoch: 300
start_epoch: 0

// cross training
view: 'joint'

// log
save_interval: 5
eval_interval: -1

knn_interval: 5
knn_k: [10, 20, 40]
knn_classes: 60
knn_t: 0.1
KNN_show: [50, 100, 150, 200, 250, 300]

**To get linear evaluation results（NTU-RGBD60 xview）**
[10.02.22|10:19:48] Eval epoch: 150
[10.02.22|10:19:54]     eval_mean_loss: 0.7034628312732722
[10.02.22|10:19:54]     Top1: 78.27%
[10.02.22|10:19:54]     Top5: 96.89%
[10.02.22|10:19:55]     Best Top1: 78.42%
[10.02.22|10:19:55] current 78.27%, best 78.42%
[10.02.22|10:19:55] The model has been saved as work_dir/linear_evaluation /skeletonclr_swap/202210011046/xview/epoch150_acc78.27_model.pt

**In your paper, NTU - RGBD60 xview acc is 84.4%，This is what reason, can you tell me what skills， thank you very much**

**Comment by czhaneva** [comment]
I think you should try the provided config file first without any modification.
I have achieved similar results with the default configs on a new server, recently.
